controllers/csi/driver/mount.go

```diff
@@ -31,12 +31,29 @@ type Mount struct {
 	ReadOnly bool
 }
 
+type bindOptions struct {
+	rootDir string
+	mounts  []Mount
+	mounter mount.Interface
+}
+
 func BindMount(rootDir string, mnts ...Mount) error {
 	if err := os.MkdirAll(rootDir, 0750); err != nil {
 		return fmt.Errorf("failed to create root directory %s: %w", rootDir, err)
 	}
```

Comment:
This `rootDir` option is redundant. Directories should be created (and removed in `BindUnmount`) based on information in `mnts`.

Reply:
I don't get how it is redundant, could you elaborate on that?

Reply:
Directories can be created when iterating over `mnts`.

Additionally, the order of entries in `mnts` is crucial. The first one is actually the root directory and all others depend on existing directories in the root (agent) directory. This code should not depend on that.

Anyways, this can be addressed in a PR on its own.